lib/platform-stats.js

```diff
@@ -165,29 +271,40 @@ export const mapParticipantsToIds = async (pgClient, participantsSet) => {
     INSERT INTO participants (participant_address)
     SELECT UNNEST($1::TEXT[]) AS participant_address
     ON CONFLICT(participant_address) DO UPDATE
-      -- this no-op update is needed to populate "RETURNING id"
+      -- this no-op update is needed to populate "RETURNING id, participant_address"
       SET participant_address = EXCLUDED.participant_address
-    RETURNING id
+    RETURNING id, participant_address
   `, [
     newAddresses
   ])
 
   assert.strictEqual(created.length, newAddresses.length)
-  return ids.concat(created.map(r => r.id))
+  // eslint-disable-next-line camelcase
+  for (const { id, participant_address } of created) {
+    participantMap.set(participant_address, id)
+  }
+
+  return participantMap
+}
+
+export const periodicDatabaseRefresh = async (createPgClient) => {
+  const pgClient = await createPgClient()
+  try {
+    await updateTopMeasurementParticipants(pgClient)
+    await aggregateAndCleanupRecentData(pgClient)
+    await updateMonthlyActiveStationCount(pgClient)
+  } finally {
+    await pgClient.end()
+  }
```

Comment:
This implementation has the problem that if one function fails, the ones following won't run.

If we want these functions to run in series, something like this could work:

```js
let err

try {
  await fnA()
} catch (_err) {
  err = _err
}

// ...

await pgClient.end()
if (err) {
  throw err
}
```

If parallel execution is ok, this simpler version works:

```js
try {
  await Promise.allSettled([
    fnA(),
    // ...
  ])
} finally {
  await pgClient.end()
}
```

Reply:
Ah good catch. I was planning to use Promise.allSettled before committing but it slipped my mind